src/server/services/decos/decos-service.ts

```diff
@@ -182,107 +191,133 @@ async function transformDecosZaakResponse(
-      vergunning,
+  if (decosZaakTransformer.afterTransform) {
+    decosZaak = await decosZaakTransformer.afterTransform(
+      decosZaak,
       decosZaakSource,
       {
         fetchDecosWorkflowDate: fetchWorkflowDate,
-        zaakTypeTransformer,
+        decosZaakTransformer,
       }
     );
   }
 
-  return vergunning;
+  return decosZaak;
 }
 
-async function transformDecosZakenResponse(
+async function transformDecosZakenResponse<
+  T extends DecosZaakTransformer<any>,
+  DZ extends DecosZaakBase = NestedType<T>,
+>(
   requestID: RequestID,
+  decosZaakTransformers: T[],
   decosZakenSource: DecosZaakSource[]
 ) {
-  const zakenToBeTransformed = [];
-
+  const zakenToBeTransformed: [T, DecosZaakSource][] = [];
   for (const decosZaakSource of decosZakenSource) {
     const zaakType = getDecosZaakTypeFromSource(decosZaakSource);
-    const zaakTypeTransformer = decosZaakTransformers[zaakType];
+    const decosZaakTransformer = decosZaakTransformers.find(
+      (transformer) => transformer.caseType == zaakType
+    );
 
-    if (!zaakTypeTransformer) {
-      captureMessage(`Decos: ${zaakType} transformer not implemented`);
+    // exclude decosZaakSources that do not have a matching decosZaakTransformer
+    if (!decosZaakTransformer) {
       continue;
     }
 
     // Exclude zaken that match the following conditions
-    if (isExcludedFromTransformation(decosZaakSource, zaakTypeTransformer)) {
+    if (isExcludedFromTransformation(decosZaakSource, decosZaakTransformer)) {
       continue;
     }
 
-    zakenToBeTransformed.push(decosZaakSource);
+    zakenToBeTransformed.push([decosZaakTransformer, decosZaakSource]);
   }
 
-  let vergunningen: Array<VergunningV2 | null> = [];
+  let decosZaken: Array<DZ | null> = [];
 
   try {
-    vergunningen = await Promise.all(
-      zakenToBeTransformed.map((decosZaak) => {
-        return transformDecosZaakResponse(requestID, decosZaak);
+    decosZaken = await Promise.all(
+      zakenToBeTransformed.map(([decosZaakTransformer, decosZaak]) => {
+        return transformDecosZaakResponse(
+          requestID,
+          [decosZaakTransformer],
+          decosZaak
+        );
       })
     );
   } catch (err) {
     captureException(err);
   }
 
-  return vergunningen
-    .filter(
-      (vergunning: VergunningV2 | null): vergunning is VergunningV2 =>
-        vergunning !== null
-    )
+  return decosZaken
+    .filter((decosZaak: DZ | null): decosZaak is DZ => decosZaak !== null)
     .sort(sortAlpha('identifier', 'desc'));
 }
 
-async function getZakenByUserKey(requestID: RequestID, userKey: string) {
-  const selectFieldsAllCases = Object.keys(SELECT_FIELDS_TRANSFORM_BASE);
-  const additionalSelectFields = Object.values(decosZaakTransformers).flatMap(
-    (zaakTransformer) => zaakTransformer.addToSelectFieldsBase ?? []
+async function getZakenByUserKey(
+  requestID: RequestID,
+  userKey: string,
+  zaakTypeTransformers: Pick<
+    DecosZaakTransformer<any>,
```

Comment:
Je kunt deze ook van `any` ontdoen...

```suggestion
  zaakTypeTransformers: Pick<
    DecosZaakTransformer<DecosZaakBase>,
```

staat er een paar keer in.